### DISCUSSION TITLE
support handling of placeholder allele name

### DISCUSSION THREAD

**Discussion by mmelchers** [Anthony-Nolan/Atlas#657] [Feature Requests]
Background: now that more and more organisations are recruiting donors in newer populations and a lot of the typing is performed using NGS, more and more alleles are discovered ever week. Some of these alleles are also present in patients that are looking for stem cells from a donor or cord. Because of that, the situation regularly happens where patients are known to have an allele that is not yet in the official nomenclature. In some cases this allele is not in the same G/P group as a known allele. There currently is no good way to input this HLA. Obviously the new allele name (if known at all) is not yet supported, but inputting it as a different allele (group) also has its issues. For example if the new typing will be named A*01:678:01:01 (once it is official) and it is not in any known G or P groups then: 

-entering A*01:XX will make the matching engine work with the KNOWN alleles and haplotypes that exist within that low resolution "group". This will lead to incorrect matching probabilities. It may even come up with full matches that just are not there. 
-you cannot enter a G, g or P group because it is not a member of one of those groups, or it is not know to the search coordinator which group it is part of.
-entering typing that is valid but different from reality is a possible quality control issue. 

Some systems handle this by allowing for the typing "NEW" e.g. A*NEW
This should not be confused with the MAC code NEW (e.g. A*01:NEW) which means "15:01/15:01N/15:12/15:14/15:19/15:26N/15:27/15:28/15:32/15:33/15:34/15:35/15:38/15:46/15:60"

This "NEW" placeholder (or any other suitable placeholder) would be nice to support because it allows for more accurate matching probabilities results. For this to work properly the behaviour should be as follows: 

any <placeholder> allele should mismatch any other allele, even the ones who have the same placeholder, because it can be a different recently identified allele. 

Therefore when a patient or donor has a <placeholder> allele in their typing, it will only be appear in a mismatch search.    

Would this be possible to implement?

**Comment by benbelow** [comment]
@mmelchers to confirm, would this approach be expected to only allow for accurate matching probabilities at *other loci*? 

i.e. if there's a `NEW` allele at A, the zero mismatch probability at A would always be 0%, so the only interesting info we can gather is the 1/2 mismatch probability at A, as well as the full probabilities at other loci?

**Reply by mmelchers** [reply]
@benbelow correct!

**Comment by benbelow** [comment]
I've raised https://github.com/Anthony-Nolan/Atlas/issues/785 to cover this, whic we can build upon as this feature gets prioritised and worked on...

**Reply by zabeen** [reply]
#785 closed as duplicate

**Comment by zabeen** [comment]
Issue raised, see https://github.com/Anthony-Nolan/Atlas/issues/876
